From b1b04be5b5312321b0cec9a017d9dd6842776ba7 Mon Sep 17 00:00:00 2001
From: Hin-Tak Leung <htl10@users.sourceforge.net>
Date: Fri, 22 Nov 2024 02:47:20 +0000
Subject: [PATCH 7/7] set c++ 17 - src/xml/SkDOM.cpp does not build with c++23;
 "std::unique_ptr<SkDOMParser> fParser;" line in header

c++ 17 is what skia recommends, the rest of webkitgtk builds with c++23.
unique_ptr itself changed behavior in c++23.

Signed-off-by: Hin-Tak Leung <htl10@users.sourceforge.net>
---
 Source/ThirdParty/skia/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Source/ThirdParty/skia/CMakeLists.txt b/Source/ThirdParty/skia/CMakeLists.txt
index 947690c9..99f0c16f 100644
--- a/Source/ThirdParty/skia/CMakeLists.txt
+++ b/Source/ThirdParty/skia/CMakeLists.txt
@@ -115,6 +115,7 @@ if (DEVELOPER_MODE AND (SKIA_ROOT_DIR OR NOT "$ENV{SKIA_ROOT_DIR}" STREQUAL ""))
     return ()
 endif ()
 
+set (CMAKE_CXX_STANDARD 17)
 #
 # The target definition lists sources which always get built.
 # The rest may be added later as-needed with target_sources().
-- 
2.47.0

